src/test/java/com/wikia/webdriver/common/logging/Log.java

```diff
@@ -0,0 +1,338 @@
+package com.wikia.webdriver.common.logging;
+
+import static org.apache.commons.lang.StringEscapeUtils.escapeHtml;
+
+import com.wikia.webdriver.common.core.CommonUtils;
+import com.wikia.webdriver.common.core.WikiaWebDriver;
+import com.wikia.webdriver.common.core.annotations.RelatedIssue;
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.imageutilities.Shooter;
+import com.wikia.webdriver.common.core.url.UrlBuilder;
+import com.wikia.webdriver.common.driverprovider.DriverProvider;
+
+import lombok.Getter;
+import net.lightbody.bmp.core.har.Har;
+import net.lightbody.bmp.core.har.HarEntry;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.assertj.core.util.Throwables;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.TimeZone;
+
+public class Log {
+  private static final String POLISH_DATE_FORMAT = "dd/MM/yyyy HH:mm:ss ZZ";
+  private static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ssZZ";
+  private static final String REPORT_PATH = "." + File.separator + "logs" + File.separator;
+  private static final String SCREEN_DIR_PATH = REPORT_PATH + "screenshots" + File.separator;
+  private static final String SCREEN_PATH = SCREEN_DIR_PATH + "screenshot";
+  private static final String LOG_FILE_NAME = "log.html";
+  public static final String LOG_PATH = REPORT_PATH + LOG_FILE_NAME;
+  private static final ArrayList<Boolean> LOGS_RESULTS = new ArrayList<>();
+  private static long imageCounter;
+
+  @Getter
+  private static boolean testStarted = false;
+
+  public static void clearLogStack() {
+    LOGS_RESULTS.clear();
+  }
+
+  private static String getPageSource(WebDriver driver) {
+    return driver.getPageSource().replaceAll("<script", "<textarea style=\"display: none\"><script")
+        .replaceAll("</script", "</script></textarea");
+  }
+
+  public static void log(String command, String description, boolean success, WebDriver driver) {
+    LOGS_RESULTS.add(success);
+    imageCounter += 1;
+
+    try {
+      new Shooter().savePageScreenshot(Log.SCREEN_PATH + Log.imageCounter, driver);
+      CommonUtils.appendTextToFile(Log.SCREEN_PATH + Log.imageCounter + ".html",
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal ".html" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)